### PR TITLE
treewide: quote urls according to rfc 0045

### PR DESCRIPTION
**Pull request by xaverdh**
###### Motivation for this change
rfc 0045

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/livedl/default.nix [hunk 1]
```diff
@@ -13,7 +13,7 @@ buildGoModule rec {
 
   sourceRoot = "source/src";
 
-  vendorSha256 = sha256:g5Y1IH1U1zOOHygTzAJuBnUj+MyPe64KHTYikipt3TY=;
+  vendorSha256 = "sha256:g5Y1IH1U1zOOHygTzAJuBnUj+MyPe64KHTYikipt3TY=";
```
Comment (thiagokokada) [inline pkgs/tools/misc/livedl/default.nix:16]: Never knew this was possible.

**Review by thiagokokada — APPROVED** [review]

**Review by r-burns — CHANGES_REQUESTED** [review]
I think you edited some node-packages.nix files which are autogenerated. These should be addressed in upstream node2nix instead

**Comment by xaverdh** [comment]
> I think you edited some node-packages.nix files which are autogenerated. These should be addressed in upstream node2nix instead

Upstream `node2nix` does quote urls since version 1.9.0 (which is the default in nixpkgs now), so when that file is regenerated next time, the homepage url will be quoted automatically.

**Comment by xaverdh** [comment]
I don't want to regenerate the file in this pr, because that might cause a rebuild of the package with newer versions as a side effect.

**Comment by r-burns** [comment]
Ah, thank you for the correction. Sounds good then.

**Review by r-burns — APPROVED** [review]

**Comment by xaverdh** [comment]
The same situation with the `php-packages` in this pr by the way, in case somebody wonders.

File: pkgs/tools/misc/livedl/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/tools/misc/livedl/default.nix:16]:
```suggestion
  vendorSha256 = "g5Y1IH1U1zOOHygTzAJuBnUj+MyPe64KHTYikipt3TY=";
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by xaverdh** [comment]
thanks!

**Pull request merged**